dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/Provisioner.java

```diff
@@ -270,6 +274,36 @@ public boolean handle(FeatureResource fr) {
                     // Get the product name for which the bundles are being installed.
                     String productName = bundleRepositoryHolder.getFeatureType();
 
+                    // If a bundle dependency specifies a required-osgi-ee capability, ensure it is satisfied before installing the bundle
+                    final String requiredOSGiEE = fr.getRequiredOSGiEE();
+                    if (requiredOSGiEE != null) {
+                        Collection<BundleCapability> foundEEs = fwkWiring.findProviders(new Requirement() {
+                            @Override
+                            public Resource getResource() {
+                                return null;
+                            }
+
+                            @Override
+                            public String getNamespace() {
+                                return ExecutionEnvironmentNamespace.EXECUTION_ENVIRONMENT_NAMESPACE;
+                            }
+
+                            @Override
+                            public Map<String, String> getDirectives() {
+                                return Collections.singletonMap(Namespace.REQUIREMENT_FILTER_DIRECTIVE, requiredOSGiEE);
+                            }
+
+                            @Override
+                            public Map<String, Object> getAttributes() {
+                                return Collections.emptyMap();
+                            }
+                        });
+                        if (foundEEs.isEmpty()) {
+                            // require-osgi-ee capability not satisfied, do not install
+                            return true;
+                        }
+                    }
+
```

Comment:
I would like to see the code to check for required OSGi EEs split out into a separate method.  This bundleList.foreach code is already too long in my opinion.  So we can reduce the code in this method to something like this:

```
                    if (missingRequiredOSGiEE(fr)) {
                            // require-osgi-ee capability not satisfied, do not install
                            return true;
                    }
```

Reply:
agreed